sips/sip-02x/sip-02x-non-sequential-multisig-transactions.md

```diff
@@ -0,0 +1,160 @@
+# Preamble
+
+SIP Number: 02x
+
+Title: Non-sequential Multisig Transactions
+
+Authors: Jeff Bencin <jbencin@hiro.so>, Vlad Bespalov <vlad@asigna.io>
+
+Consideration: Technical
+
+Type: Consensus
+
+Status: Draft
+
+Created: 2023-08-30
+
+License: BSD 2-Clause
+
+Sign-off:
+
+Discussions-To: https://github.com/stacksgov/sips
+
+# Abstract
+
+This SIP proposes a new multisig transaction format which is intended to be easier to use than the current format described in SIP-005.
+It does not remove support for the current format, rather it is intended to co-exist with the old format and give users a choice of which format to use.
+
+The issue with the current format is that it establishes a signer order when funds are sent to multisig account address, and requires signers to sign in the same order to spend the funds.
+In practice, the current format has proven difficult to understand and implement, as evidenced by the lack of Stacks multisig implementations today.
+
+This new format intends to simplify the signing algorithm and remove the requirement for in-order signing, without comprimising on security or increasing transaction size.
+It is expected that this will lead to better wallet support for Stacks multisig transactions.
+
+# Introduction
+
+Currently, a multisig transaction requires the first signer to sign the transaction itself, and subsequent signers to sign the signature of the previous signer.
+For a transaction with *n* signers, the final signature is generated in the following way:
+
+```
+signature_n(...(signature_2(signature_1(tx))))
+```
+
+There are a few drawbacks to doing it this way:
+
+- The order in which the signers must sign is fixed as soon as funds are sent to a multisig account, which limits flexibility when creating a spending transaction from a multisig account
+- The process of signing a transaction requires each signer to validate the entire signature chain before signing, in order to make sure it matches the transaction, leading to `O(n^2)` signing times
```

Comment:
Slight correction -- it requires each signer to perform `O(n)` signature verifications.  The `O(n^2)` total verifications is distributed across `n` signers.

Reply:
Yes, I'll clarify